root/static/scripts/edit/URLCleanup.js

```diff
@@ -3871,7 +3871,25 @@ const CLEANUPS: CleanupEntries = {
   },
   'soundcloud': {
     match: [new RegExp('^(https?://)?([^/]+\\.)?soundcloud\\.com', 'i')],
-    restrict: [LINK_TYPES.soundcloud],
+    restrict: [
+      LINK_TYPES.soundcloud,
+      {
+        recording: LINK_TYPES.downloadfree.recording,
+        release: LINK_TYPES.downloadfree.release,
+      },
+      {
+        recording: LINK_TYPES.downloadpurchase.recording,
+        release: LINK_TYPES.downloadpurchase.release,
+      },
+      {
+        recording: LINK_TYPES.streamingfree.recording,
+        release: LINK_TYPES.streamingfree.release,
+      },
+      {
+        recording: LINK_TYPES.streamingpaid.recording,
+        release: LINK_TYPES.streamingpaid.release,
+      },
+    ],
```

Comment:
It should be possible to combine (free or paid) downloading and streaming too. Untested suggestion:

```suggestion
    restrict: [
      LINK_TYPES.soundcloud,
      {
        recording: LINK_TYPES.downloadfree.recording,
        release: LINK_TYPES.downloadfree.release,
      },
      {
        recording: LINK_TYPES.downloadpurchase.recording,
        release: LINK_TYPES.downloadpurchase.release,
      },
      {
        recording: LINK_TYPES.streamingfree.recording,
        release: LINK_TYPES.streamingfree.release,
      },
      {
        recording: LINK_TYPES.streamingpaid.recording,
        release: LINK_TYPES.streamingpaid.release,
      },
      {
        recording: [
          LINK_TYPES.downloadfree.recording,
          LINK_TYPES.streamingfree.recording,
        ],
        release: [
          LINK_TYPES.downloadfree.release,
          LINK_TYPES.streamingfree.release,
        ],
      },
      {
        recording: [
          LINK_TYPES.downloadpurchase.recording,
          LINK_TYPES.streamingpaid.recording,
        ],
        release: [
          LINK_TYPES.downloadpurchase.release,
          LINK_TYPES.streamingpaid.release,
        ],
      },
    ],
```

Reply:
Hmm. Does this otherwise block entering combinations? I guess I didn't test it well enough, if so... and it seems like a strange way of implementing it.

Reply:
Yes, it otherwise blocks entering combinations. I even tested it locally to make sure.

`restrict` has been designed to be used like that, even though there is no precedent implementation of optional combination. The only combination so far is for `jamendo` but it is auto-selected as all recordings are free to download and to stream.

Here is the only code documentation about `restrict` to my knowledge: https://github.com/metabrainz/musicbrainz-server/blob/v-2021-11-15/root/static/scripts/edit/URLCleanup.js#L376-L381